src/components/Forms/Configs/KubernetesConfigs.tsx

```diff
@@ -0,0 +1,403 @@
+import { Formik, Field, Form } from "formik";
+import { useState } from "react";
+import { Button } from "../../Button";
+import { CodeEditor } from "../../CodeEditor";
+import { Switch } from "../../Switch";
+import { Tab, Tabs } from "../../Tabs/Tabs";
+import { TextInput } from "../../TextInput";
+import YAML from "yaml";
+import { isEmpty } from "lodash";
+import { ConfigArrayFormField } from "./ConfigArrayFormField";
+import ConfigListFormField from "./ConfigListFormField";
+
+type KubernetesConfigsProps = {
+  canEdit?: boolean;
+  loadSpec?: () => Record<string, any>;
+  getSpec?: (spec: string) => void;
+  specFormat?: "yaml" | "json";
+};
+
+export default function KubernetesConfigs({
+  canEdit = true,
+  loadSpec = () => ({}),
+  getSpec = () => {},
+  specFormat = "yaml"
+}: KubernetesConfigsProps) {
+  const [activeTabs, setActiveTabs] = useState<"Form" | "Code">("Form");
+
+  const initialValues: Record<string, any> = loadSpec();
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={(values, actions) => {
+        const spec =
+          specFormat === "yaml"
+            ? YAML.stringify(values)
+            : JSON.stringify(values);
+        getSpec(spec);
+      }}
+    >
+      {({ values, setFieldValue, setValues }) => (
+        <Form className="flex flex-col">
+          <Tabs
+            activeTab={activeTabs}
+            onSelectTab={(v) => setActiveTabs(v as "Code" | "Form")}
+          >
+            <Tab label="Form" value="Form">
+              <div className="flex flex-col space-y-4 p-4">
+                <div className="flex flex-col space-y-2">
+                  <label className="font-semibold">BaseScraper</label>
+                  <div className="flex flex-col p-4 space-y-2 border border-gray-200 rounded-md">
```

Comment:
This form should not include any boilerplate styling as it will be difficult to maintain across all forms - Maybe we just return a list of fields ?